app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+class User < ApplicationRecord
+  validates :name, presence: true,
+    length: { maximum: Settings.validations.name.max_length }
+
+  validates :email, presence: true,
+    length: { maximum: Settings.validations.email.max_length },
+    format: { with: Settings.validations.email.regex },
+    uniqueness: true
+
+  validates :password, presence: true,
+    length: { minimum: Settings.validations.password.min_length }
+
+  has_secure_password
+
+  before_save { email.downcase! }
```

Comment:
please write method for email.downcase!  and call it here like sample_app

Reply:
done